lib/preview/previewUtils.js

```diff
@@ -95,14 +97,70 @@ const silenceConsoleWhile = async (act, ...args) => {
   return result;
 }
 
+const memoize = (func, cacheBustCallback) => {
+  const cache = {};
+  return async (...args) => {
+    const stringArgs = args.toString();
+    const storedResult = cache[stringArgs];
+    if (storedResult && (cacheBustCallback ? !cacheBustCallback() : true)) {
+      //console.log(`Cache hit ${func}`)
+      return storedResult;
+    }
+    //console.log(`Cache miss ${func}`)
+    const res = await func(...args);
+    cache[stringArgs] = res;
+    return res;
+  }
+}
+
+const hidePreviewInDest = (previewDest) => previewDest.split('/').slice(2).join('/');
+
+const trackPreviewEvent = async (action) => {
+  if (!isTrackingAllowed()) {
+    return;
+  }
+  const accountId = getAccountId();
+
+  trackUsage(
+    'cli-interaction',
+    'INTERACTION',
+    {
+      applicationName: 'hubspot.preview',
+      os: `${platform()} ${release()}`,
+      authType: getAuthType(accountId),
+      action,
+    },
+    accountId
+  ).catch(
```

Comment:
I just have this under `cli-interaction` since it's part of that product (whereas VSCE is its own). What exactly we track here can be changed, I think `applicationName: 'hubspot.preview'` and the individual actions is a good start though.